### PR TITLE
Clean up 'shell=True' in using subprocess module

### PR DESCRIPTION
**Pull request by wangqr**
Currently, `configure.py` always use the running python interpreter to find the site-packages path. When user has specified a different python in the previous step, the specified one should be used.

### CODE REVIEW

**Comment by yifeif** [comment]
Thanks for the PR @wangqr. #12238 just got merged with a similar fix. I saw you also had other clean-ups in your PR. Would you mind merge your change to the current head?

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please.

**Comment by wangqr** [comment]
Now this PR mainly do the `shell=True` cleanup, which I am still working. Sorry for pushing the commit before done.

**Comment by yifeif** [comment]
no worries. Let me know when it is ready for review again. Thanks!

**Comment by wangqr** [comment]
@yifeif Now I am done. Thanks!

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please.

File: configure.py [hunk 1]
```diff
@@ -707,10 +713,11 @@ def get_native_cuda_compute_capabilities(environ_cp):
   """
   device_query_bin = os.path.join(
       environ_cp.get('CUDA_TOOLKIT_PATH'), 'extras/demo_suite/deviceQuery')
-  cmd = (r'"%s" | grep "Capability" | grep -o "[0-9]*\.[0-9]*" | sed '
-         '":a;{N;s/\\n/,/};ba"') % device_query_bin
   try:
-    output = run_shell(cmd)
+    output = run_shell(device_query_bin).split('\n')
```
Comment (yifeif) [inline configure.py]: Looks like it is not happy that it cannot find deviceQuery. https://ci.tensorflow.org/job/tensorflow-pull-requests-gpu/6417/console
Can we add a check for that if we are not using shell=True?

**Comment by wangqr** [comment]
Should have fixed the command not found problem.

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

**Pull request merged**